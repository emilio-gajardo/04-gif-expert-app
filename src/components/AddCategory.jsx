import { useState } from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prop-types
export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;
        // setCategories((categories) => [inputValue, ...categories]);
        setInputValue('');
        onNewCategory(inputValue.trim());
    }

    return (
        <form onSubmit={(event) => onSubmit(event)} aria-label='form'>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={(event) => onInputChange(event)}
            />
        </form>
    );
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}