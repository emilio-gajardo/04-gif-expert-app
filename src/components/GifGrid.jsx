import PropTypes from 'prop-types';
/* eslint-disable react/prop-types */
import { GifItem } from './GifItem.jsx';
import { useFetchGifs } from '../hooks/useFetchGifs.js'

export const GifGrid = ({ category }) => {

    // Custom Hook
    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>

            {(isLoading) && (<h3>Cargando...</h3>)}

            <div className='card-grid'>
                {
                    images.map((image) => (
                        <GifItem key={image.id} {...image} />
                    ))
                }
            </div>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}