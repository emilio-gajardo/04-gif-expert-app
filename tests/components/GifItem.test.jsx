import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Test: <GiItem />', () => {

    const altTitle = 'Title testing';
    const srcUrl = 'https://media4.giphy.com/media/cXblnKXr2BQOaYnTni/giphy.gif';

    test('1) <img/> debe contener el title', () => {
        render(<GifItem title={altTitle} url={srcUrl} />);
        const { alt } = screen.getByRole('img');
        expect(alt).toBe(altTitle);
    });

    test('2) <img/> debe contener el src', () => {
        render(<GifItem title={altTitle} url={srcUrl} />);
        const { src } = screen.getByRole('img');
        expect(src).toBe(srcUrl);
    });

    test('3) <img/> debe coincidir el snapshot', () => {
        const { container } = render(<GifItem title={altTitle} url={srcUrl} />);
        // screen.debug();
        expect(container).toMatchSnapshot();
    });

    test('4) Debe renderizar el componente con el title', () => {
        render(<GifItem title={altTitle} url={srcUrl} />);
        expect(screen.getByText(altTitle)).toBeTruthy();
    });
});