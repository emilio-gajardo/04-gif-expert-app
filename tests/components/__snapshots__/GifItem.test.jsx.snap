// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: <GiItem /> 3) <img/> debe coincidir el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title testing"
      src="https://media4.giphy.com/media/cXblnKXr2BQOaYnTni/giphy.gif"
    />
    <p>
      Title testing
    </p>
  </div>
</div>
`;
